### PR TITLE
Infra 844 signalfx batching filter

### PR DESCRIPTION
**Pull request by nathanleiby**
Issue: https://clever.atlassian.net/browse/INFRA-844

Filter to support batching messages before sending to SignalFx API.

### CODE REVIEW

**Comment by nathanleiby** [comment]
Note: there's a modules file - I'm experimenting with being able abstact to shared module (e.g. we have used `sub_func` in influx, stathat, and signalfx). I'm able to do it locally but need to see how it works given the `package.path` at runtime in Heka.

File: lua/modules/util.lua [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    Payload = true,
+    Hostname = true,
+    Pid = true,
+    Logger = true,
+    Severity = true,
+    EnvVersion = true
+}
+
+-- Used for interpolating message fields into series name.
+function M.sub_func(key)
```
Comment (mohit) [inline lua/modules/util.lua]: a better name than `sub_func`

File: lua/filters/signalfxbatch.lua [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+    Hostname = true,
+    Pid = true,
+    Logger = true,
+    Severity = true,
+    EnvVersion = true
+}
+
+-- Allows interpolating message fields into a string.
+-- Used to make custom metric names.
+local function sub_func(key)
```
Comment (mohit) [inline lua/filters/signalfxbatch.lua]: seems like it's in modules?

File: lua/filters/signalfxbatch.lua [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+local value_field = read_config("value_field") or "value"
+local dimensions_str = read_config("dimensions") or ""
+local msg_type = read_config("msg_type") or "signalfxbatch"
+
+local use_subs
+if string.find(metric_name, "%%{[%w%p]-}") then
+    use_subs = true
+end
+
+local base_fields_map = {
```
Comment (mohit) [inline lua/filters/signalfxbatch.lua]: also why both here and in modules

File: lua/filters/signalfxbatch.lua [hunk 4]
```diff
@@ -0,0 +1,117 @@
+require "cjson"
```
Comment (mohit) [inline lua/filters/signalfxbatch.lua]: docs?

**Comment by mohit** [comment]
ping

File: lua/filters/signalfxbatch.lua (hunk 4 above)
Comment (nathanleiby) [inline lua/filters/signalfxbatch.lua]: Added docs  c27c0e0

File: lua/filters/signalfxbatch.lua (hunk 3 above)
Comment (nathanleiby) [inline lua/filters/signalfxbatch.lua]: For now, removed import of Lua `modules`. It has some implications for how we run Heka (need to mount different directories). 

I'll update to use them in a future PR.

**Pull request merged**